### ISSUE TITLE
Allow @Queue annotation to allow variables

### ISSUE THREAD

**Issue by charlie-harvey** [micronaut-projects/micronaut-jms#323]
### Feature description

```kotlin
@JMSListener(CONNECTION_FACTORY_BEAN_NAME)
class TextConsumer {

    val messages: MutableList<String> = Collections.synchronizedList(ArrayList())

    @Queue(value = "${some.awesome.queue.name}", concurrency = "1-5")
    fun receive(@MessageBody body: String) {
        messages.add(body)
    }
}
```

Is this possible? We thought it was a really good idea when we started using AWS to prefix all our queue names with the env name. Even though they are in different environments. So I want my queues to be called "dev-theQueue" and "prod-theQueue". If its not possible then its not possible. Just thought I'd ask.

Thanks.

**Comment by windhamg** [comment]
Same issue. Having to hardwire the queue name in the annotation definitely reduces the usefulness.

**Comment by windhamg** [comment]
> Same issue. Having to hardwire the queue name in the annotation definitely reduces the usefulness.

Actually, I found this was due to my own error. I'm using Groovy and the use of double-quotes (`"`) around the property placeholder expressions was causing Groovy to interpret it as a GString and causing a static type-checking error, e.g.:
```groovy
@Queue(value = "${some.awesome.queue.name}", concurrency = "1-5")
```
Once I changed to single quotes (`'`), or escaped the `$`, the issue was resolved and the queue name was resolved from the property. I believe Kotlin would have the same issue, @charlie-harvey.
